geode-core/src/main/java/org/apache/geode/internal/JarDeployer.java

```diff
@@ -158,61 +156,55 @@ protected File getNextVersionedJarFile(String unversionedJarName) {
    * @return The version number embedded in the filename
    */
   public static int extractVersionFromFilename(final String filename) {
-    final Matcher matcher = versionedPattern.matcher(filename);
+    final Matcher matcher = SEQUENCED_VERSION_SCHEME.matcher(filename);
     if (matcher.find()) {
       return Integer.parseInt(matcher.group(2));
     } else {
       return 0;
     }
   }
 
-  protected Set<String> findDistinctDeployedJarsOnDisk() {
-    // Find all deployed JAR files
-    final File[] oldFiles =
-        this.deployDirectory.listFiles((file, name) -> versionedPattern.matcher(name).matches());
-
-    // Now add just the original JAR name to the set
-    final Set<String> jarNames = new HashSet<>();
-    for (File oldFile : oldFiles) {
-      Matcher matcher = versionedPattern.matcher(oldFile.getName());
-      matcher.find();
-      jarNames.add(matcher.group(1) + ".jar");
-    }
-    return jarNames;
+  public static boolean isSequenceVersion(String filename) {
+    return SEQUENCED_VERSION_SCHEME.matcher(filename).find();
+  }
+
+  public static boolean isSemanticVersion(String filename) {
+    return SEMANTIC_VERSION_SCHEME.matcher(filename).find();
   }
 
   /**
-   * Find all versions of the JAR file that are currently on disk and return them sorted from newest
-   * (highest version) to oldest
+   * get the artifact id from the existing files on the server. This will skip files that
+   * do not have sequence id or semantic version appended to them.
    *
-   * @param unversionedJarName Name of the JAR file that we want old versions of
-   * @return Sorted array of files that are older versions of the given JAR
+   * @param versionedJarFileName: the file names that exists on the server, could be file with
+   *        sequence numbers abc.v1.jar, or file with semantic version abc-1.0.0.jar
```

Comment:
these javadocs are wrong. "abc-1.0.0.jar" should now also have a sequence number.